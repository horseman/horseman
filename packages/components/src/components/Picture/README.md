This renders a picture element with `<srcset>` and a fallback `<img>` tag.
Props not otherwise specified will be passed to the underlying `<img>` tag.

### Examples
```js
const imageData = {
  "src": "http://placehold.it/800x200",
  "alt": '',
  "srcset": [
    {
      "src": "http://placehold.it/320x200",
      "width": 320,
      "height": 200
    },
    {
      "src": "http://placehold.it/720x200",
      "width": 720,
      "height": 200
    },
  ]
};

<Picture imageData={imageData} />
```
