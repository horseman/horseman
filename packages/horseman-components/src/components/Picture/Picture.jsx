import React from "react";
import styled from "styled-components";

import { imageType } from "../../types";

const StyledPicture = styled.picture`
  img {
    max-width: 100%;
    width: 100%;
    height: auto;
    display: block;
  }
`;

const Picture = ({ src, srcset, alt, ...rest }) => {
  const items = srcset.map(sourceSet => (
    <source
      key={sourceSet.src}
      media={`(min-width:${sourceSet.width}px)`}
      srcSet={sourceSet.src}
    />
  ));

  return (
    <StyledPicture>
      {items}
      <img srcSet={src} alt={alt} {...rest} />
    </StyledPicture>
  );
};

Picture.defaultProps = {
  alt: " ",
  srcset: [],
};

Picture.propTypes = {
  ...imageType,
};

/**
 * @component
 */
export default Picture;
