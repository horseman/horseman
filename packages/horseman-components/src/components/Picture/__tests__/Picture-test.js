import React from "react";
import renderer from "react-test-renderer";

import Picture from "../";

describe("Picture", () => {
  test("renders without srcset", () => {
    const tree = renderer.create(<Picture src="foo" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders responsive with srcset", () => {
    const tree = renderer
      .create(
        <Picture
          src="foo"
          srcset={[
            {
              src: "foo",
              width: 300,
            },
            {
              src: "bar",
              width: 600,
            },
          ]}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
