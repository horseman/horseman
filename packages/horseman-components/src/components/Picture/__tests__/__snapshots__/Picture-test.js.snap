// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture renders responsive with srcset 1`] = `
.c0 img {
  max-width: 100%;
  height: auto;
  display: block;
}

<picture
  className="c0"
>
  <source
    media="(min-width:300px)"
    srcSet="foo"
  />
  <source
    media="(min-width:600px)"
    srcSet="bar"
  />
  <img
    alt=" "
    srcSet="foo"
  />
</picture>
`;

exports[`Picture renders without srcset 1`] = `
.c0 img {
  max-width: 100%;
  height: auto;
  display: block;
}

<picture
  className="c0"
>
  <img
    alt=" "
    srcSet="foo"
  />
</picture>
`;
